Training/Program.cs

```diff
@@ -1,7 +1,74 @@
+// ---------------------------------------------------------------------------------------
+// Training ~ A training program for new joinees at Metamation, Batch- July 2023.
+// Copyright (c) Metamation India.
+// ---------------------------------------------------------------------------
+// Program.cs
+// Program to Implement a Stack<T> class using arrays as the underlying data structure.
+// The Stack<T> should start with an initial capacity of 4 and double its capacity when needed.
+// class TStack<T> {
+//    public void Push (T item) { }
+//    public T Pop () { }
+//    public T Peek () { }
+//    public bool IsEmpty { get; }
+// }
+// ---------------------------------------------------------------------------------------
+
 namespace Training {
-    internal class Program {
-        static void Main (string[] args) {
-            Console.WriteLine ("Hello, World!");
-        }
-    }
+   #region TStack<T> ---------------------------------------------------------------------
+   class TStack<T> {
+      #region Properties --------------------------------------------
+      public int Capacity => mData.Length;
+      public int Count => mSize;
+
+      /// <summary>Indicates whether the stack is empty</summary>
+      public bool IsEmpty => mData.Length == 0;
+      #endregion
+
+      #region Methods ----------------------------------------------------------------------
+      /// <summary>Gets the top element from the stack and returns it</summary>
+      public T Pop () {
+         if (mSize == 0) throw new InvalidOperationException ("Stack is empty");

```

Comment:
Since this code is common for both Pop and Peek, move it to a separate function.